server/channels/store/sqlstore/channel_store_categories.go

```diff
@@ -457,20 +462,40 @@ func (s SqlChannelStore) completePopulatingCategoryChannelsT(db dbSelecter, cate
 	}
 
 	// A subquery that is true if the channel does not have a SidebarChannel entry for the current user on the current team
-	doesNotHaveSidebarChannel := sq.Select("1").
-		Prefix("NOT EXISTS (").
-		From("SidebarChannels").
-		Join("SidebarCategories on SidebarChannels.CategoryId=SidebarCategories.Id").
-		Where(sq.And{
-			sq.Expr("SidebarChannels.ChannelId = ChannelMembers.ChannelId"),
-			sq.Eq{"SidebarCategories.UserId": category.UserId},
-			sq.Eq{"SidebarCategories.TeamId": category.TeamId},
-		}).
-		Suffix(")")
+	var doesNotHaveSidebarChannel sq.SelectBuilder
+	if isMySQL {
+		doesNotHaveSidebarChannel = sq.Select("/*+ QB_NAME(subq1) */ 1").
+			Prefix("NOT EXISTS (").
```

Comment:
Technically I could have refactored out the Prefix and Suffix as well. But I feel it's more readable this way. 0/5.